tracing/src/instrument.rs

```diff
@@ -120,6 +120,8 @@ pub trait WithSubscriber: Sized {
     }
 }
 
+impl<T: Sized> WithSubscriber for T {}
```

Comment:
The bound here is copied from [`tracing-futures`](https://docs.rs/tracing-futures/0.2.5/tracing_futures/trait.WithSubscriber.html#impl-WithSubscriber). It might be worth adding a `Future` bound to avoid polluting autocompletes too much though?

Reply:
The `tracing-futures` version does not have a `Future` bound because it also allows wrapping types implementing other traits, such as `Stream`, with a subscriber. However, `tracing`'s version only provides implementations for `Future`, because `tracing` depends only on the standard library, not the `futures` crate. So, it should be fine to restrict the blanket impl to `T: Future`, here.

However, when the `Stream` trait eventually does make it into `std`, we'll probably want to add a `Stream` impl for WithSubscriber<T: Stream>`, as well. Then, we'll have to make the blanket impl more general. Technically, I think this _is_ a breaking change, as it could break user `WithSubscriber` impls for types that don't implement 'Future`. To protect against that, I think we would need to [seal](https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed) the trait so that downstream code can't add `WithSubscriber` impls. Then, it would be fine to add a `Future` bound on the blanket impl, for now.